user-guide/Advanced_Modules/User Defined APIs/UD_APIs_Triggering_an_API.md

```diff
@@ -0,0 +1,153 @@
+
+```
+Bearer 24te8bRBOhrDj6TkXkim4qh9GpDXsdGudib57j7C/CU=
+```
+
+### Content-Type
+
+In case the API needs input from the user using the HTTP body, you need to specify a Content-Type header.
+
+The endpoint supports the following content types:
+
+- application/json
+- application/xml
+- text/xml
+- text/plain
+
+> [!NOTE]
+> Adding no Content-Type header to a request is also supported. In this case, the body is expected to be empty.
+
+### Content-Length
+
+The Content-Length header should be calculated and filled in automatically depending on how you send the request. It contains the length of the body in bytes. It is important that this is correct, as *UserDefinableApiEndpoint* will only read the number of bytes of the body specified in the Content-Length header.
+
+> [!IMPORTANT]
+> The endpoint limits the size of requests to 30 MB.
+
+### Body
+
+An API can expect input. This input is passed in the body. The format of the input is defined in the definition of the API. If the API definition is set to accept parameters, these are expected to be passed as JSON in a key-value format. See [User input data](xref:UD_APIs_Define_New_API#user-input-data).
+
+The body is expected to be encoded in UTF-8.
+
+## The response
+
+The API returns an HTTP status code indicating the status of the request and a body.
+
+In case some input from the user is missing, or the user sends a request with a wrong HTTP method, the API will return an HTTP status code indicating the error and a JSON body with more information.
+
+The endpoint itself can also return [errors](#errors) with corresponding status code to indicate something went wrong before the script was executed.
+
+The API will always return responses encoded in UTF-8.
+
+### Errors
+
+The endpoint can return an error if something goes wrong. An error will be returned in the following format, with an HTTP response code indicating that something went wrong:
+
+```json
+{
+    "errors": [
+        {
+            "title": "Error occurred while handling the request, contact your admin with the provided errorCode and faultingNode ID.",
+            "detail": "Invalid json body received.",
+            "errorCode": 10,
+            "faultingNode": 123
+        }
+    ]
+}
+```
+
+For now, the `errors` object will always contain only one error, but this can change in the future.
+
+> [!NOTE]
+> We recommend that API maintainers return errors in the same format for the sake consistency (but without *errorCode*). To know what HTTP status code to return, take a look [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status).
+
+#### detail
+
+The *detail* field of an error contains a client-safe message explaining what went wrong.
+
+#### errorCode
+
+The *errorCode* field of an error contains an error code that can be used by the API maintainer to find out what went wrong:
+
+> [!IMPORTANT]
+> Do not share the explanation of these ErrorCodes with the API consumers. These are for internal use only.
```

Comment:
I guess we can. This was initially added because there were different error codes for entering an invalid secret, an existing secret that is not valid on this API, or a secret that does not exist. These differences could be abused to bruteforce the secret, but all of the above have been replaced by AuthenticationFailed